### PR TITLE
UrbanSound8K

### PR DESCRIPTION
**Pull request by justinsalamon**
### Description

This PR adds a dataloader and index for the UrbanSound8K dataset: https://urbansounddataset.weebly.com/urbansound8k.html

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of data included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [ ] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [ ] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

### CODE REVIEW

**Comment by justinsalamon** [comment]
@magdalenafuentes this PR is not ready yet, but since I only half know what I'm doing here it would be great if you could give `urbansound8k.py` a quick look to make sure I'm going in the right direction (and if not, course correct!).

I hope to have this PR completed tomorrow.

**Comment by justinsalamon** [comment]
p.s. this PR is based off of #4 so it looks like there are tons of changes but in reality the only new thing is the `urbansound8k.py` file. It'll look cleaner once we've merged #4 into master and into this branch.

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py]: nit: id of the clip

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py]: This feels a bit long here, and since the info is already in the docs I'd simplify it

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py]: Same here

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py]: Same here

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py:249]: This is good!

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py:254]: Do we need jams at all in soundata?

File: soundata/datasets/urbansound8k.py
Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py]: fixed

File: soundata/datasets/urbansound8k.py
Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py]: Done, replaced everything after the first line with: "Please see the Dataset Info in the soundata documentation for further details."

File: soundata/datasets/urbansound8k.py
Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py]: I've updated this docstring quite a bit please have another look once the PR is ready for another review

File: soundata/datasets/urbansound8k.py
Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py:249]: ok cool!

File: soundata/datasets/urbansound8k.py
Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py:254]: I think we do, it's used for a variety of DCASE dataset (e.g. augmented urbansound8k, URBAN-SED, FUSS). In fact maybe JAMS is being used more for DCASE research right now than for MIR :sweat_smile:

Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py:254]: Also everything generated by Scaper...

Comment (magdalenafuentes) [reply soundata/datasets/urbansound8k.py:254]: Oh! Well, that's great then that we have the `jams_converter` implemented. We might have to give it a look to adapt it more for how it's been used in DCASE

Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py:254]: 👍

**Comment by justinsalamon** [comment]
@magdalenafuentes ready for review!

File: docs/index.rst
Comment (magdalenafuentes) [inline docs/index.rst:55]: Good catch!

File: scripts/make_urbansound8k_index.py
Comment (magdalenafuentes) [inline scripts/make_urbansound8k_index.py]: nit: I think we should import this from `validate` instead of adding this code to each `make_index_dataset.py` we write

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py]: We can remove these

File: soundata/datasets/urbansound8k.py
Comment (magdalenafuentes) [inline soundata/datasets/urbansound8k.py:262]: 👍

**Review by magdalenafuentes — APPROVED** [review]
Niiiice! Just left a few minor comments and almost good to go! Yay!

**Comment by magdalenafuentes** [comment]
Just three things left: add to `soundata.rst`, run `test_full_dataset` and add `test_urbansound8k`. We'll add the last one after `rc` release so @pseeth can get started, but goes without much warranties ;)

File: scripts/make_urbansound8k_index.py
Comment (justinsalamon) [reply scripts/make_urbansound8k_index.py]: agreed, fixed

File: soundata/datasets/urbansound8k.py
Comment (justinsalamon) [reply soundata/datasets/urbansound8k.py]: done

**Pull request merged**
